keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt

```diff
@@ -0,0 +1,132 @@
+package com.netflix.spinnaker.keel.slack.callbacks
+
+import com.netflix.spinnaker.keel.api.constraints.ConstraintStatus
+import com.netflix.spinnaker.keel.core.api.parseUID
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.slack.SlackService
+import com.netflix.spinnaker.kork.exceptions.SystemException
+import com.slack.api.model.block.LayoutBlock
+import com.slack.api.model.kotlin_extension.block.withBlocks
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.time.Instant
+
+/**
+ * This handler will handel slack callbacks coming from Manual Judgment notifications.
+ * First, it will update the constraint status based on the user response (either approve/reject)
+ * Second, it will construct an updated notification with the action preformed and the user how did it.
+ */
+@Component
+class ManualJudgmentCallbackHandler(
+  private val clock: Clock,
+  private val repository: KeelRepository,
+  private val slackService: SlackService
+) {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  //Updating the constraint status based on the user response (either approve/reject)
+  fun handleMJResponse(slackCallbackResponse: SlackCallbackHandler.SlackCallbackResponse) {
+    val constraintUid = slackCallbackResponse.constraintId
+    val currentState = repository.getConstraintStateById(parseUID(constraintUid))
+      ?: throw SystemException("constraint@callbackId=$constraintUid", "constraint not found")
+
+    val user = slackService.getEmailByUserId(slackCallbackResponse.user.id)
+    val actionStatus = slackCallbackResponse.actions.first().value
+
+    log.debug(
+      "Updating constraint status based on notification interaction: " +
+        "user = $user, status = $actionStatus}"
+    )
+
+    repository
+      .storeConstraintState(
+        currentState.copy(
+          status = ConstraintStatus.valueOf(actionStatus),
+          judgedAt = Instant.now(),
+          judgedBy = user
+        )
+      )
+
+    //convert status to actual action --> like OVERRIDE_PASS to "approve"
+    val actionPerformed = actionsMap[actionStatus]
+    if (actionPerformed == null) {
+      log.warn("can't map slack action status to an actual action and therefore can't send an updated notification.")
+      return
+    }
```

Comment:
Could this ever happen in practice since we convert the `actionStatus` to a `ConstraintStatus` on line 46, which would fail with an exception if it's an unknown value? BTW, you could use `ConstraintStatus` values as the keys to `actionsMap`.

Reply:
I'm not sure if it can happen or not, but when converting the status to the actual actions (like `approve` and so on) with the `actionsMap`, it can be nullable, so I thought to add a check for it.

Reply:
Ok.